dhcpv4/nclient4/lease.go

```diff
@@ -0,0 +1,64 @@
+//This is lease support for nclient4
+
+package nclient4
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+//Lease contains a DHCPv4 lease after DORA.
+//note: Lease doesn't include binding interface name
+type Lease struct {
+	Offer        *dhcpv4.DHCPv4
+	ACK          *dhcpv4.DHCPv4
+	CreationTime time.Time
+	IDOptions    dhcpv4.Options //DHCPv4 options to identify the client like client-id, option82/remote-id
+}
+
+// WithClientIDOptions configures a list of DHCPv4 option code that DHCP server
+// uses to identify client, beside the MAC address.
+func WithClientIDOptions(cidl dhcpv4.OptionCodeList) ClientOpt {
+	return func(c *Client) (err error) {
+		c.clientIDOptions = cidl
+		return
+	}
+}
+
+//Release send DHCPv4 release messsage to server, based on specified lease.
+//release is sent as unicast per RFC2131, section 4.4.4.
+//Note: some DHCP server requries of using assigned IP address as source IP,
+//use nclient4.WithUnicast to create client for such case.
+func (c *Client) Release(lease *Lease) error {
+	if lease == nil {
+		return fmt.Errorf("lease is nil")
+	}
+	modList := []dhcpv4.Modifier{
```

Comment:
I think part of this should be moved so that it can be reused.

I think we should have a `NewReleaseFromACK` in `dhcpv4.go`, similar to [NewRequestFromOffer](https://github.com/insomniacslk/dhcp/blob/master/dhcpv4/dhcpv4.go#L235)

Maybe at another time?

Reply:
ok, I could move this to NewReleaseFromACK